WordPress/Classes/System/RootViewPresenter+MeNavigation.swift

```diff
@@ -17,23 +17,26 @@ extension RootViewPresenter {
 
     /// access to sub levels
     func navigateToAccountSettings() {
-        showMeScene(animated: false) { meViewController in
-            self.popMeTabToRoot()
-            meViewController?.navigateToAccountSettings()
+        showMeScreen()
+        popMeTabToRoot()
+        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(250)) {
+            self.meViewController?.navigateToAccountSettings()
```

Comment:
I believe this `asyncAfter` is because you wanted the pop animation to finish before executing the Account Settings navigation.

💡 If that's the case, I suggest the following code that achieves the same thing.

```swift
CATransaction.perform {
    popMeTabToRoot()
} completion: {
    self.meViewController?.navigateToAccountSettings()
}
```

The completion block will be executed as soon as the pop animation finishes, and we don't have to guess the pop animation duration ( 250ms ).

Reply:
Hey @salimbraksa - I tried using `CATransaction`, but I can't get the completion block to execute.

I actually need the `showMeScreen()` animation to finish (i.e. pushing the Me screen onto the nav stack) before navigating to the subscreens. The `popMeScreenToRoot()` implementation is empty for the simplified app UI type (see: `MySitesCoordinator+RootViewPresenter.swift`). 

```
CATransaction.perform {
    showMeScreen()
    popMeScreenToRoot()
} completion: {
    self.meViewController?.navigateToAccountSettings()
}
```

For context, the delay is needed for the simplified app UI type. Normal / static screens don't need the delay.

Reply:
Hey @momo-ozawa I tried to look into why the `CATransaction` didn't work, and I found something interesting. 

I refactored the `navigateToAccountSettings` method to this.

```swift
    func navigateToAccountSettings() {
        CATransaction.perform {
            showMeScreen()
            popMeTabToRoot()
        } completion: {
            self.meViewController?.navigateToAccountSettings()
        }
    }
```

Then, I triggered the deep link using this command:

```
xcrun simctl openurl booted https://wordpress.com/me/account
```

The "Me" screen did show up, and the `CATransaction` completion handler was executed but the **App Settings did not appear.** 

I reverted `navigateToAccountSettings` back to its previous implementation and tested the deep linking again. And same as before, the **App Settings did not appear**.

After some investigation, I found out that 2 `MeViewController` instances are created, and the one in `self.meViewController?.navigateToAccountSettings()` is not the one shown on screen.

![CleanShot 2023-09-25 at 19 00 03@2x](https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/1b6cbca2-4f6a-446b-8f31-c97cdc7d2fe3)

Reply:
The weird thing is that I remember the first time I tested this PR, the deep linking to **App Settings** did work, but now it doesn't.

Reply:
Both issues were not fixed @momo-ozawa ( at least on my side ). The "Account Settings" navigation is not working and 2 `MeViewController` instances are created.

| Account Settings | 2 MeViewController Instances |
| ---------------- | ------------------------------- |
| <video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/ac583902-12a8-4c1a-8cc3-00be64948639"/> | <img width="1034" alt="CleanShot 2023-09-27 at 13 40 58@2x" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/fc68f9da-16b6-42c6-984e-899973e85067">|

@kean Are you able to reproduce this issue?

Reply:
@salimbraksa Is this happening just for Jetpack + iPhone?


https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/7428263b-42a1-486f-adc3-0811dd1d4e0a

Reply:
@momo-ozawa I haven't tested the other cases yet, but this bug seems to be related to the `meViewController` property in `WPTabViewController`. It looks like the `meViewController` property isn't the same as the `MeViewController` embedded in the tab.

![CleanShot 2023-09-27 at 14 33 45@2x](https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/17df3a96-6a19-40f3-a53d-a4a2094934cf)

Reply:
Also, when I run the app, the `MeViewController` initializer is called twice:

| 1st call | 2nd call |
| ------ | --------- |
| <img width="1038" alt="CleanShot 2023-09-27 at 14 38 50@2x" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/76769f67-4b21-4dc9-94f9-5718bdc59a77"> | <img width="1038" alt="CleanShot 2023-09-27 at 14 39 47@2x" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/d5473bff-aedb-4335-9893-7bef8c7d90bf"> |

Reply:
> @kean Are you able to reproduce this issue?

I'm able to reproduce it with the latest commit. 

I think it has to do with state restoration. If I remove `extension MeViewController: UIViewControllerRestoration`, it works fine.

I'd suggest reviewing the codebase and removing state native restoration (`UIViewControllerRestoration`) everywhere because it's been broken for years now and is always a major source of defects.

Reply:
I originally tested on a clean build where there was no restorable state. That's likely why it worked then.

Reply:
This is the root cause of the issue @momo-ozawa. The restoration logic replaces the Me screen with a new instance, so the `meViewController` property in `WPTabBarController` is not used anymore.

![CleanShot 2023-09-27 at 15 18 23@2x](https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/0f3891ae-6dc3-43b5-b82d-c30e36813245)

Reply:
@momo-ozawa Removing the state restoration logic fixes the issue. I also tested the `CATransaction` in `navigateToAccountSettings` and it worked. Can you give it another try?

```swift
    func navigateToAccountSettings() {
        CATransaction.perform {
            showMeScreen()
        } completion: {
            self.meViewController?.navigateToAccountSettings()
        }
    }
```

Reply:
Hey @salimbraksa, while CATransaction works for the normal / static screen app UI types (i.e. WordPress, static screens enabled OR Jetpack), it still doesn't work for the simplified app UI type (i.e. WordPress, phase four enabled)


https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/21f87bc4-3823-4680-ba11-9c1bd28bcc17


https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/e7bca091-ab60-4ca2-9f2f-9897ae67072d

Reply:
Let's keep this simple and not make the `CATransaction` change for now since that's what we've already tested and proved to work. It looks OK from the user perspective.

Reply:
> it still doesn't work for the simplified app UI type (i.e. WordPress, phase four enabled)

`CATransaction` not working could be a symptom of something else, because I don't see why it shouldn't work. Anyhow, I'll test the simplified UI type and investigate with `CATransaction` didn't work. 🙇

Reply:
It's a good suggestion. Maybe would try it in `trunk`? This PR targets the release, and it'll be great to merge it sooner than later.

Reply:
@kean @momo-ozawa I've opened a [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/21648) fixing the `CATransaction`, and it's targeting this PR.